client/components/calypso-i18n-provider/index.tsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
+import { I18nProvider } from '@automattic/react-i18n';
+import i18n from 'i18n-calypso';
+
+const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
+	const [ localeData, setLocaleData ] = React.useState( i18n.state.locale );
+
+	useEffect( () => {
+		const onChange = () => {
+			const nextLocaleData = i18n.state.locale;
+			setLocaleData( nextLocaleData );
+		};
+
+		i18n.on( 'change', onChange );
+		return () => {
+			i18n.off( 'change', onChange );
+		};
+	}, [] );
+
+	useEffect( () => {
+		setWpI18nLocaleData( localeData );
```

Comment:
Can `setWpI18nLocaleData` be called in the `onChange` handler? As it is now, is it possible that the rerender triggered by `i18n.change` and `setState` happens before `setWpI18nLocaleData` is called, and therefore works with old translations?